MekHQ/docs/DistributedMekHQ.md

```diff
@@ -0,0 +1,310 @@
+
+The Host Campaign is always Team 1.
+
+This message is only sent by the Host Campaign.
+
+#### DailyLog
+The DailyLog message notifies the Host Campaign of a daily log activity.
+
+The DailyLog message has the following properties:
+- Campaign UUID
+- Log Date
+- Log List
+    - Log Entry
+
+This message may be sent at any point in a day and may be sent mulitple times, but implementations could delay sending this message until a day has completed.
+
+This message is redistributed to all Remote Campaigns on the same team as the sending Campaign.
+
+#### WireTransferRequest
+The WireTransferRequest message notifies the Host Campaign of an intent to perform a wire transfer between two Campaigns.
+
+The WireTransferRequest message has the following properties:
+- Source Campaign UUID
+- Target Campaign UUID
+- Request UUID
+- Money Amount
+
+The amount may be positive (debiting the source campaign) or negative (debiting the target campaign).
+
+The source Campaign must move any debited funds into a Pending account and may not use those funds until the transfer has been completed.
+
+This message is redistributed only to the target Campaign.
+
+#### WireTransferResponse
+The WireTransferResponse message notifies the Host Campaign of the response to a wire transfer request.
+
+The WireTransferResponse message has the following properties:
+- Source Campaign UUID
+- Target Campaign UUID
+- Request UUID
+- Accepted?
+
+The target Campaign must send a rejection response if the request if it has insufficient funds to process a debit.
+
+The target Campaign must move any debited funds into a Pending account and may not use those funds until the transfer has been completed.
+
+The Host Campaign must send a rejection WireTransferCompleted message if the request is not recognized.
+
+This message is only consumed by the Host Campaign.
+
+#### WireTransferCompleted
+The WireTransferCompleted message notifies two participating Campaigns that their wire transfer has been accepted by the Host.
+
+The WireTransferCompleted message has the following properties:
+- Source Campaign UUID
+- Target Campaign UUID
+- Request UUID
+- Accepted?
+
+This message must be rejected by the source or target Campaigns if they do not recognize it.
+
+This message is sent to the originating source and target Campaigns upon receipt of a WireTransferResponse.
+
+#### TradeRequest
+The TradeRequest message notifies the Host Campaign of an intent to trade between two campaigns.
+
+The TradeRequest message has the following properties:
+- Source Campaign UUID
+- Target Campaign UUID
+- Request UUID
+- ?
+
+Details TBD
+
+If the Source and Target Campaign already have an active trade, the Host Campaign shall send a rejection TradeCompleted to the Source Campaign.
+
+This message is redistributed to the target campaign by the Host Campaign.
+
+#### CancelTradeRequest
+The CancelTradeRequest message notifies the Host Campaign that a trade partner has cancelled a request.
```

Comment:
This should only be allowed if neither party has sent any of their resources out, and would need to be enforced. Otherwise, I'd imaging shenanigans could easily occur.

Reply:
Resources are in 'escrow' until the trade is accepted and the host notifies campaigns of its acceptance.

Reply:
Perfect.